aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -508,6 +509,15 @@ Tensor & scatter_cpu_(Tensor & self, int64_t dim, const Tensor & index, const Te
   return self;
 }
 
+Tensor & scatter_cpu_reduce_(Tensor & self, int64_t dim, const Tensor & index,
+                      const Tensor & src, std::string reduce) {
+  TORCH_CHECK(reduce != "sum" || reduce != "subtract" ||
+              reduce != "multiply" || reduce != "divide",
+              "reduce argument must be either of add, subtract, multiply or divide.");
```

Comment:
Please, fix to `reduce =='sum' || reduce ==...`.

Reply:
Also, trim `reduce` and convert it to lower case.

Reply:
There is an idea. You create a hash table of names and in TORCH_CHECK do the set inclusion test.

Reply:
I'm also not very keen on string manipulation in the hot path.  Maybe ultimately it would make sense to replace it with an enum. In any case, even if user-facing api has strings, this function ideally should be the last place where strings are used.

Reply:
Yes. I did have that mind. Was waiting for approval on the overall design without macros. Will push changes.